pkg/bosh/converter/container_factory.go

```diff
@@ -479,6 +481,11 @@ func bpmProcessContainer(
 		container.Env = append(container.Env, corev1.EnvVar{Name: name, Value: value})
 	}
 
+	// Arbitrary envs should override container.Env.
+	if len(arbitraryEnvs) != 0 {
+		container.Env = overrideEnv(container.Env, arbitraryEnvs)
```

Comment:
In some sense, this is still mutating `container.Env`. Ideally, you wouldn't set `container.Env = <something>`, but rather initialize the `corev1.Container{}` above with the envs already. What you can do is to combine this with the logic of the `range process.Env` to be done before `container` is instantiated.